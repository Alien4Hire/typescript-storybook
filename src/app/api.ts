import {
  BaseQueryFn,
  FetchArgs,
  FetchBaseQueryError,
} from '@reduxjs/toolkit/query';

import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { logout, getCaTokenCookie } from './reducers/login/loginSlice';
import { setToastNotification } from './reducers/appSlice/appSlice';

export const baseUrl = process.env.REACT_APP_API || '/api/v1';

declare module '@reduxjs/toolkit/query' {
  interface FetchBaseQueryError {
    status: number;
    data:
    | {
      id?: string;
      location?: string;
      message: string;
      path?: string;
      status?: number;
      type?: string;
    }[]
    | {
      id?: string;
      location?: string;
      message: string;
      path?: string;
      status?: number;
      type?: string;
    };
  }
}

const baseQuery = fetchBaseQuery({
  baseUrl,
  prepareHeaders: (headers, { getState }) => {
    const token = getCaTokenCookie();
    if (token) {
      headers.set('authorization', `Bearer ${token}`);
    }
    headers.set('Content-Type', 'application/json;charset=UTF-8');
    return headers;
  },
});

const baseQueryWithErrorHandling: BaseQueryFn<
  string | FetchArgs,
  unknown,
  FetchBaseQueryError
> = async (args, api, extraOptions) => {
  let result = await baseQuery(args, api, extraOptions);
  if (result.error) {
    if (result.error.status === 401) {
      api.dispatch(logout());
      /**
       * Instead of logging out when the token is invalid, it is possible to refresh the token here.
       */
      // try to get a new token
      // const refreshResult = await baseQuery('/refreshToken', api, extraOptions);
      // if (refreshResult.data) {
      //   // store the new token
      //   api.dispatch(setNewToken(refreshResult.data));
      //   // retry the initial query
      //   result = await baseQuery(args, api, extraOptions);
      // } else {
      //   api.dispatch(logout());
      // }
    } else {
      const message = Array.isArray(result.error?.data)
        ? result.error?.data[0]?.message
        : result.error?.data?.message;
      api.dispatch(
        setToastNotification({
          type: 'error',
          open: true,
          message,
        })
      );
    }
  }
  return result;
};

export const apiBase = createApi({
  reducerPath: 'api',
  baseQuery: baseQueryWithErrorHandling,
  //List of tag associated with our data endpoints
  tagTypes: [
    'Templates',
    'ActivitySettings',
    'TemplateViewTypes',
    'ProductModels',
    'LearningObjectives',
    'Taxonomies',
    'Activities',
    'Lessons',
    'Content',
  ],
  refetchOnFocus: true,
  refetchOnReconnect: true,
  refetchOnMountOrArgChange: true,
  endpoints: () => ({}),
});
