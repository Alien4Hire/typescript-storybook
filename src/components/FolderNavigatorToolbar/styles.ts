import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
