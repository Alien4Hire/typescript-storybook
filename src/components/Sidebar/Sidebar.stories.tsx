import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Sidebar from './Sidebar';

export default {
  title: 'Component/Sidebar',
  component: Sidebar,
} as ComponentMeta<typeof Sidebar>;

const Template: ComponentStory<typeof Sidebar> = (args) => (
  <Router>
    <Sidebar {...args} />
  </Router>
);
export const SidebarView = Template.bind({});
