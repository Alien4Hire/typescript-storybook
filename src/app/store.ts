import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { combineReducers } from '@reduxjs/toolkit';
import configReducer from './reducers/config/configSlice';
import loginReducer from './reducers/login/loginSlice';
import activitySettingsReducer from './reducers/activitySettings/activitySettingsSlice';
import lessonsReducer from './reducers/lessons/lessonsSlice';
import searchReducer from './reducers/search/searchSlice';
import resourceFilterReducer from './reducers/resourcesFilter/resourcesFilterSlice';
import templateReducer from './reducers/templates/templateSlice';
import appReducer from './reducers/appSlice/appSlice';
import templateViewTypesReducer from './reducers/templateViewTypes/templateViewTypesSlice';
import targetFolderSideBarReducer from './reducers/targetFolderSideBar/targetFolderSideBarSlice';

import { setupListeners } from '@reduxjs/toolkit/query/react';
import { apiBase } from './api';

const rootReducer = combineReducers({
  config: configReducer,
  login: loginReducer,
  activitySettings: activitySettingsReducer,
  lessons: lessonsReducer,
  templates: templateReducer,
  search: searchReducer,
  resourcesFilter: resourceFilterReducer,
  app: appReducer,
  templateViewTypes: templateViewTypesReducer,
  targetFolderSideBar: targetFolderSideBarReducer,
  [apiBase.reducerPath]: apiBase.reducer,
});

// Use also for store setup in jest unit tests
// Use-case: unit test needs preloadedState
// Example: client/src/components/Pages/LoginPage/LoginPage.test.tsx
export const setupStore = (preloadedState: object = {}) => {
  return configureStore({
    reducer: rootReducer,
    devTools: process.env.NODE_ENV !== 'production',
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(apiBase.middleware),
    preloadedState,
  });
};

export const store = setupStore();

setupListeners(store.dispatch);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
