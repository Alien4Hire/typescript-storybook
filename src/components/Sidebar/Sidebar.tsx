import { useLocation, useNavigate, useParams } from 'react-router-dom';

import { List } from '@mui/material';

import { useAppSelector } from '../../app/hooks';
import { getSideBarStatus } from '../../app/reducers/appSlice/appSlice';

import {
  Line,
  StyledDrawer,
  StyledHome,
  StyledSearch,
  StyledListIcon,
  StyledCog,
  StyledFile,
  StyledButton,
} from './Sidebar.styles';
import { theme } from '../../theme';

export type MlSidebarPropsT = {
  disabled?: boolean;
  active?: number;
};

type ListItemPropsT = {
  to: string;
  children: JSX.Element;
  title?: string;
};

const Sidebar = ({ disabled = false }: MlSidebarPropsT) => {
  const navigate = useNavigate();
  const location = useLocation();

  const sideBarStatus = useAppSelector(getSideBarStatus);

  const ListItem = ({ to, children, title }: ListItemPropsT) => (
    <StyledButton
      onClick={() => navigate(to)}
      active={to === location.pathname ? 1 : 0}
      variant="text"
      disabled={disabled}
      theme={theme}
      title={title}
      aria-label={`button ${title}`}
    >
      {children}
    </StyledButton>
  );
  // const template = useAppSelector(selectCurrentTemplate);
  const { templateId, templateVersionId } = useParams();

  return (
    <StyledDrawer
      open={sideBarStatus}
      theme={theme}
      variant="permanent"
      anchor="left"
    >
      <List>
        <ListItem to="/">
          <StyledHome title="Template List" />
        </ListItem>
        <Line theme={theme} />
        <ListItem to={`/templates/${templateId}/search/${templateVersionId}`}>
          <StyledSearch title="Search Content" />
        </ListItem>
        <ListItem
          to={`/templates/${templateId}/activities/${templateVersionId}`}
        >
          <StyledListIcon title="Resources" />
        </ListItem>
        <ListItem title="Project Plan" to="/templates">
          <StyledFile />
        </ListItem>
        <Line theme={theme} />
        <ListItem to={`/templates/${templateId}/edit/${templateVersionId}`}>
          <StyledCog title="Edit Settings" />
        </ListItem>
      </List>
    </StyledDrawer>
  );
};

export default Sidebar;
