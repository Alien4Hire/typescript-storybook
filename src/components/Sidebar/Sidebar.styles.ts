import { Drawer } from '@mui/material';
import { css, styled } from '@mui/material/styles';

import IconButton from '@mui/material/IconButton';

import { ReactComponent as Home } from '../../icons/home.svg';
import { ReactComponent as Search } from '../../icons/search.svg';
import { ReactComponent as ListIcon } from '../../icons/list_left.svg';
import { ReactComponent as Cog } from '../../icons/cog.svg';
import File from '@mui/icons-material/InsertDriveFileOutlined';

export const Line = styled('div')(({ theme }) => ({
  border: `1px solid ${theme.palette.gray.dark}`,
  marginTop: '10px',
  marginBottom: '10px',
  position: 'relative',
}));

export const StyledButton = styled(IconButton)(({ theme, ...props }) => ({
  background: props.active && theme.palette.primary.main + ' !important',
  margin: '0px !important',
  height: '48px !important',
  width: '100%',
  borderRadius: '0px !important',
  boxSizing: 'border-box !important',
  '& svg': {
    fill: props.active && theme.palette.white.main,
    color: props.active && theme.palette.white.main,
  },
}));

export const StyledDrawer = styled(Drawer)(({ theme }) => ({
  '& .MuiDrawer-paper': {
    width: '72px',
    marginTop: '64px',
    height: 'calc(100% - 64px)',
    background: theme.palette.gray.veryDark,
    border: 0,
  },
}));

const sharedStyle = css`
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
`;

export const StyledHome = styled(Home)`
  ${sharedStyle}
`;

export const StyledSearch = styled(Search)`
  ${sharedStyle}
`;

export const StyledListIcon = styled(ListIcon)`
  ${sharedStyle}
`;

export const StyledCog = styled(Cog)`
  ${sharedStyle}
`;

export const StyledFile = styled(File)`
  ${sharedStyle}
`;
