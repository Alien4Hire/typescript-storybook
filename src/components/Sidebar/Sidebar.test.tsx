import { render } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { Provider } from 'react-redux';
import Sidebar from './Sidebar';
import { store } from '../../app/store';
import { theme } from '../../theme';
import '../../setupTests';

const baseProps = {
  disabled: false,
};

const renderComponent = () => {
  return render(
    <Provider store={store}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>
          <Router>
            <Sidebar {...baseProps} />
          </Router>
        </ThemeProvider>
      </StyledEngineProvider>
    </Provider>
  );
};

describe('<Sidebar />', () => {
  it('renders without crashing', () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });
});
