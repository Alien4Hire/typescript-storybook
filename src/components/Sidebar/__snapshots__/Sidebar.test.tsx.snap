// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c13 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c2 {
  background-color: #ffffff;
  color: rgba(0,0,0,0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  box-shadow: none;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c5::-moz-focus-inner {
  border-style: none;
}

.c5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c6 {
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0,0,0,0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c6:hover {
  background-color: rgba(0,0,0,0.04);
}

.c6.Mui-disabled {
  background-color: transparent;
  color: rgba(0,0,0,0.26);
}

.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  z-index: 1200;
  -webkit-overflow-scrolling: touch;
  position: fixed;
  top: 0;
  outline: 0;
  left: 0;
  border-right: 1px solid rgba(0,0,0,0.12);
}

.c4 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c9 {
  border: 1px solid #686868;
  margin-top: 10px;
  margin-bottom: 10px;
  position: relative;
}

.c7 {
  background: #00758e !important;
  margin: 0px !important;
  height: 48px !important;
  width: 100%;
  border-radius: 0px !important;
  box-sizing: border-box !important;
}

.c7 svg {
  fill: #ffffff;
  color: #ffffff;
}

.c10 {
  background: 0;
  margin: 0px !important;
  height: 48px !important;
  width: 100%;
  border-radius: 0px !important;
  box-sizing: border-box !important;
}

.c10 svg {
  fill: 0;
  color: 0;
}

.c1 .MuiDrawer-paper {
  width: 72px;
  margin-top: 64px;
  height: calc(100% - 64px);
  background: #383838;
  border: 0;
}

.c8 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

.c11 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

.c12 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

.c15 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

.c14 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

@media print {
  .c5 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c6:hover {
    background-color: transparent;
  }
}

<body>
    <div>
      <div
        class="c0 MuiDrawer-root MuiDrawer-docked c1"
      >
        <div
          class="c2 MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 c3 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
        >
          <ul
            class="c4 MuiList-root MuiList-padding"
          >
            <button
              active="1"
              aria-label="button undefined"
              class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c7"
              tabindex="0"
              type="button"
              variant="text"
            >
              <svg
                class="c8"
                title="Template List"
              >
                home.svg
              </svg>
            </button>
            <div
              class="c9"
            />
            <button
              active="0"
              aria-label="button undefined"
              class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c10"
              tabindex="0"
              type="button"
              variant="text"
            >
              <svg
                class="c11"
                title="Search Content"
              >
                search.svg
              </svg>
            </button>
            <button
              active="0"
              aria-label="button undefined"
              class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c10"
              tabindex="0"
              type="button"
              variant="text"
            >
              <svg
                class="c12"
                title="Resources"
              >
                list_left.svg
              </svg>
            </button>
            <button
              active="0"
              aria-label="button Project Plan"
              class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c10"
              tabindex="0"
              title="Project Plan"
              type="button"
              variant="text"
            >
              <svg
                aria-hidden="true"
                class="c13 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c14"
                data-testid="InsertDriveFileOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
                />
              </svg>
            </button>
            <div
              class="c9"
            />
            <button
              active="0"
              aria-label="button undefined"
              class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c10"
              tabindex="0"
              type="button"
              variant="text"
            >
              <svg
                class="c15"
                title="Edit Settings"
              >
                cog.svg
              </svg>
            </button>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": .c13 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c2 {
  background-color: #ffffff;
  color: rgba(0,0,0,0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  box-shadow: none;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c5::-moz-focus-inner {
  border-style: none;
}

.c5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c6 {
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0,0,0,0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c6:hover {
  background-color: rgba(0,0,0,0.04);
}

.c6.Mui-disabled {
  background-color: transparent;
  color: rgba(0,0,0,0.26);
}

.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  z-index: 1200;
  -webkit-overflow-scrolling: touch;
  position: fixed;
  top: 0;
  outline: 0;
  left: 0;
  border-right: 1px solid rgba(0,0,0,0.12);
}

.c4 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c9 {
  border: 1px solid #686868;
  margin-top: 10px;
  margin-bottom: 10px;
  position: relative;
}

.c7 {
  background: #00758e !important;
  margin: 0px !important;
  height: 48px !important;
  width: 100%;
  border-radius: 0px !important;
  box-sizing: border-box !important;
}

.c7 svg {
  fill: #ffffff;
  color: #ffffff;
}

.c10 {
  background: 0;
  margin: 0px !important;
  height: 48px !important;
  width: 100%;
  border-radius: 0px !important;
  box-sizing: border-box !important;
}

.c10 svg {
  fill: 0;
  color: 0;
}

.c1 .MuiDrawer-paper {
  width: 72px;
  margin-top: 64px;
  height: calc(100% - 64px);
  background: #383838;
  border: 0;
}

.c8 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

.c11 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

.c12 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

.c15 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

.c14 {
  width: 24px;
  height: 24px;
  color: #ccc;
  fill: #ccc;
  background-color: transparent;
}

@media print {
  .c5 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c6:hover {
    background-color: transparent;
  }
}

<div>
    <div
      class="c0 MuiDrawer-root MuiDrawer-docked c1"
    >
      <div
        class="c2 MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 c3 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
      >
        <ul
          class="c4 MuiList-root MuiList-padding"
        >
          <button
            active="1"
            aria-label="button undefined"
            class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c7"
            tabindex="0"
            type="button"
            variant="text"
          >
            <svg
              class="c8"
              title="Template List"
            >
              home.svg
            </svg>
          </button>
          <div
            class="c9"
          />
          <button
            active="0"
            aria-label="button undefined"
            class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c10"
            tabindex="0"
            type="button"
            variant="text"
          >
            <svg
              class="c11"
              title="Search Content"
            >
              search.svg
            </svg>
          </button>
          <button
            active="0"
            aria-label="button undefined"
            class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c10"
            tabindex="0"
            type="button"
            variant="text"
          >
            <svg
              class="c12"
              title="Resources"
            >
              list_left.svg
            </svg>
          </button>
          <button
            active="0"
            aria-label="button Project Plan"
            class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c10"
            tabindex="0"
            title="Project Plan"
            type="button"
            variant="text"
          >
            <svg
              aria-hidden="true"
              class="c13 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c14"
              data-testid="InsertDriveFileOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zM6 20V4h7v5h5v11H6z"
              />
            </svg>
          </button>
          <div
            class="c9"
          />
          <button
            active="0"
            aria-label="button undefined"
            class="c5 MuiButtonBase-root c6 MuiIconButton-root MuiIconButton-sizeMedium c10"
            tabindex="0"
            type="button"
            variant="text"
          >
            <svg
              class="c15"
              title="Edit Settings"
            >
              cog.svg
            </svg>
          </button>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
